### DISCUSSION TITLE
Is it possible to add custom headers when submitting a <Form>?

### DISCUSSION THREAD

**Discussion by nullndr** [remix-run/remix#2184] [Q&A] [answered]
I need to verify if an incoming request has an authorization header given by an external function call.

I have this little form that allow for CRUD operations, and I would love that each submit could also include the authorization header.

Is it possible?

**Comment by sergiodxa** [comment, marked as answer]
It's not possible, but it would be easier to use a cookie for the login and not keep the token on the client, cookies will go on every request automatically.

**Reply by nullndr** [reply]
I know this, but I can only generate this token client side, the function I was talking about only generate a header, but nvm.

Thank you.

**Reply by sergiodxa** [reply]
Generate the token and send it to an action with useSubmit, then your action can store the token in a cookie and from that point you can use that.

That’s how login with supabase or Firebase are typically done in Remix which also gives you the token only client side.
